yarn-project/sequencer-client/src/sequencer/public_processor.test.ts

```diff
@@ -93,13 +93,18 @@ describe('public_processor', () => {
     const publicKernelSpy = jest.spyOn(publicKernel, 'publicKernelCircuitNoInput');
     processor = new PublicProcessor(db, publicCircuit, publicKernel, publicProver, contractDataSource);
 
-    const publicCircuitOutput = makePublicCircuitPublicInputs();
-    publicCircuit.publicCircuit.mockResolvedValue(publicCircuitOutput);
-
     const path = times(PUBLIC_DATA_TREE_HEIGHT, i => Buffer.alloc(32, i));
     db.getSiblingPath.mockResolvedValue(new SiblingPath(path));
 
     const tx = makePublicTx();
+    // mock Public Circuit output
+    const publicCircuitOutput = makePublicCircuitPublicInputs();
+    publicCircuitOutput.publicCallStack = new Array(PUBLIC_CALL_STACK_LENGTH).fill(Fr.ZERO);
+    // set storageContractAddress to txRequest.to
+    publicCircuitOutput.callContext.storageContractAddress = tx.txRequest.txRequest.to;
```

Comment:
Should we push this fix to the factory itself, as you did with the others?

Reply:
nice catch